packages/host/tests/integration/components/ai-assistant-panel-test.gts

```diff
@@ -1076,61 +1076,63 @@ module('Integration | ai-assistant-panel', function (hooks) {
-      .exists(
-        "test room 3 is the most recently created room and it's opened initially",
+    try {
+      await setCardInOperatorModeState(`${testRealmURL}Pet/mango`);
+      await renderComponent(
+        class TestDriver extends GlimmerComponent {
+          <template>
+            <OperatorMode @onClose={{noop}} />
+            <CardPrerender />
+          </template>
+        },
       );
 
-    await click('[data-test-past-sessions-button]');
-    await click(`[data-test-enter-room="${room2Id}"]`);
-
-    await click('[data-test-close-ai-assistant]');
-    await click('[data-test-open-ai-assistant]');
-    await waitFor(`[data-room-settled]`);
-    assert
-      .dom(`[data-test-room="${room2Id}"]`)
-      .exists(
-        "test room 2 is the most recently selected room and it's opened initially",
+      await matrixService.createAndJoinRoom('test1', 'test room 1');
+      const room2Id = await matrixService.createAndJoinRoom(
+        'test2',
+        'test room 2',
       );
-
-    await click('[data-test-close-ai-assistant]');
-    window.localStorage.setItem(
-      currentRoomIdPersistenceKey,
-      "room-id-that-doesn't-exist-and-should-not-break-the-implementation",
-    );
-    await click('[data-test-open-ai-assistant]');
-    await waitFor(`[data-room-settled]`);
-    assert
-      .dom(`[data-test-room="${room3Id}"]`)
-      .exists(
-        "test room 3 is the most recently created room and it's opened initially",
+      const room3Id = await matrixService.createAndJoinRoom(
+        'test3',
+        'test room 3',
       );
 
-    window.localStorage.removeItem(currentRoomIdPersistenceKey); // Cleanup
+      await openAiAssistant();
+      await waitFor(`[data-room-settled]`);
+
+      assert
+        .dom(`[data-test-room="${room3Id}"]`)
+        .exists(
+          "test room 3 is the most recently created room and it's opened initially",
+        );
+
+      await click('[data-test-past-sessions-button]');
+      await click(`[data-test-enter-room="${room2Id}"]`);
+
+      await click('[data-test-close-ai-assistant]');
+      await click('[data-test-open-ai-assistant]');
+      await waitFor(`[data-room-settled]`);
+      assert
+        .dom(`[data-test-room="${room2Id}"]`)
+        .exists(
+          "test room 2 is the most recently selected room and it's opened initially",
+        );
+
+      await click('[data-test-close-ai-assistant]');
+      window.localStorage.setItem(
+        currentRoomIdPersistenceKey,
+        "room-id-that-doesn't-exist-and-should-not-break-the-implementation",
+      );
+      await click('[data-test-open-ai-assistant]');
+      await waitFor(`[data-room-settled]`);
+      assert
+        .dom(`[data-test-room="${room3Id}"]`)
+        .exists(
+          "test room 3 is the most recently created room and it's opened initially",
+        );
+    } finally {
+      window.localStorage.removeItem(currentRoomIdPersistenceKey); // Cleanup
```

Comment:
Were you seeing bleedthrough with this? I wouldn’t expect it with the window mock.

Reply:
Yeah, this test was flaky for me. Whenever we are trying to restore state outside of an after hook we should always use a finally.

Reply:
annoying! but understood, thanks.